internal/reporter.go

```diff
@@ -32,22 +32,38 @@ func (reporter reporter) Report(format string, pointLines string) (*http.Respons
 		return nil, formatError
 	}
 
-	// compress
+	buf, err := linesToBuffer(pointLines)
+	if err != nil {
+		return &http.Response{}, err
+	}
+
+	req, err := reporter.buildRequest(format, buf)
+	if err != nil {
+		return &http.Response{}, err
+	}
+
+	return reporter.execute(req)
+}
+
+func linesToBuffer(pointLines string) (bytes.Buffer, error) {
```

Comment:
I don't think bytes.Buffer supports copying by simple assignment. Instead of returning a `bytes.Buffer` or even a `*bytes.Buffer` consider instead returning a `[]byte`.  Returning a `[]byte` is more concise and less confusing because you are eagerly constructing the bytes in this function anyway.  Also consider renaming `linesToBuffer` to `linesToBytes` or something like that.

Reply:
I think what you're saying is that returning a `bytes.Buffer` is a form of copying by value and that the buffer might not end up containing the intended gzipped content. I haven't dug into that but I like the idea of returning the simpler `[]byte` type - pushing that change now.

Reply:
Slightly concerned about whether the refactor, although more readable, might cause any noticeable performance changes given we're creating a new buffer twice now (I don't know if we think of this code as performance sensitive or have any performance tests around it)